app/controllers/api/databases.php

```diff
@@ -3439,6 +3440,282 @@ function updateAttribute(
+                        $processDocument($relatedCollection, $relation);
+                    }
+                }
+            }
+        };
+
+        $processDocument($collection, $document);
+
+        $events
+            ->setParam('databaseId', $databaseId)
+            ->setParam('collectionId', $collection->getId())
+            ->setParam('documentId', $document->getId())
+            ->setContext('collection', $collection)
+            ->setContext('database', $database)
+            ->setPayload($response->output($document, Response::MODEL_DOCUMENT));
+
+        $response->dynamic($document, Response::MODEL_DOCUMENT);
+    });
+
+App::patch('/v1/databases/:databaseId/collections/:collectionId/documents/:documentId/decrementAttribute/:key')
+    ->alias('/v1/database/collections/:collectionId/documents/:documentId/decrementAttribute', ['databaseId' => 'default'])
+    ->desc('Decrement Document Attribute')
+    ->groups(['api', 'database'])
+    ->label('event', 'databases.[databaseId].collections.[collectionId].documents.[documentId].update')
+    ->label('scope', 'documents.write')
+    ->label('audits.event', 'document.update')
+    ->label('audits.resource', 'database/{request.databaseId}/collection/{request.collectionId}/document/{response.$id}')
+    ->label('usage.metric', 'documents.{scope}.requests.update')
+    ->label('usage.params', ['databaseId:{request.databaseId}', 'collectionId:{request.collectionId}'])
+    ->label('abuse-key', 'ip:{ip},method:{method},url:{url},userId:{userId}')
+    ->label('abuse-limit', APP_LIMIT_WRITE_RATE_DEFAULT * 2)
+    ->label('abuse-time', APP_LIMIT_WRITE_RATE_PERIOD_DEFAULT)
+    ->label('sdk.auth', [APP_AUTH_TYPE_SESSION, APP_AUTH_TYPE_KEY, APP_AUTH_TYPE_JWT])
+    ->label('sdk.namespace', 'databases')
+    ->label('sdk.method', 'decrementDocumentAttribute')
+    ->label('sdk.description', '/docs/references/databases/decrement-document-attribute.md')
+    ->label('sdk.response.code', Response::STATUS_CODE_OK)
+    ->label('sdk.response.type', Response::CONTENT_TYPE_JSON)
+    ->label('sdk.response.model', Response::MODEL_DOCUMENT)
+    ->label('sdk.offline.model', '/databases/{databaseId}/collections/{collectionId}/documents')
+    ->label('sdk.offline.key', '{documentId}')
+    ->param('databaseId', '', new UID(), 'Database ID.')
+    ->param('collectionId', '', new UID(), 'Collection ID.')
+    ->param('documentId', '', new UID(), 'Document ID.')
+    ->param('key', '', new Key(), 'Attribute Key.')
+    ->param('value', null, new Numeric(), 'Value to decrement attribute by')
+    ->inject('requestTimestamp')
+    ->inject('response')
+    ->inject('dbForProject')
+    ->inject('events')
+    ->inject('mode')
+    ->action(function (string $databaseId, string $collectionId, string $documentId, string $key, int|float $value, ?\DateTime $requestTimestamp, Response $response, Database $dbForProject, Event $events, string $mode) {
+        if ($value <= 0) {
```

Comment:
We can use a Range validator?

Reply:
We can, but `Range(0)` allows for 0, so we would need to check for 0 here. Would you rather me switch it to `Range()`?

Reply:
Yea forgot about the floats.
This looks bad:
new Range(0.00001, PHP_INT_MAX, \Utopia\Validator::TYPE_FLOAT)
I will resolve 👍